packaging/opae/rpm/create

```diff
@@ -66,56 +83,30 @@ do
     mkdir -p "${RPMBUILD_DIR}/$d"
 done
 
-# Create source tarball.
+# clone opae legancy and tests
+
 cd "${SDK_DIR}/.."
-tar --transform='s/opae-sdk/opae/' \
-  --exclude=.git \
-  --exclude=.gitignore \
-  --exclude=.github \
-  --exclude=.travis.yml \
-  --exclude=opae.spec.in \
-  --exclude=opae.spec \
-  --exclude=opae-libs/external \
-  --exclude=opae-libs/tests \
-  --exclude=opae-libs/plugins/ase \
-  --exclude=opae-libs/cmake/config/libopae-all.spec.in \
-  --exclude=opae-libs/cmake/config/run_coverage_test.sh.in \
-  --exclude=opae-libs/cmake/config/run_coverage_test_local.sh.in\
-  --exclude=external \
-  --exclude=platforms \
-  --exclude=samples/base \
-  --exclude=samples/hello_afu \
-  --exclude=samples/hello_mpf_afu \
-  --exclude=samples/intg_xeon_nlb \
-  --exclude=samples/base \
-  --exclude=scripts \
-  --exclude=tests \
-  --exclude=tools/fpgametrics \
-  --exclude=tools/libboard/board_dc \
-  --exclude=tools/extra/ras \
-  --exclude=tools/extra/fpgabist \
-  --exclude=tools/extra/pac_hssi_config \
-  --exclude=tools/extra/fpgadiag \
-  --exclude=tools/extra/c++utils \
-  --exclude=tools/extra/pyfpgadiag \
-  --exclude=tools/extra/pypackager \
-  --exclude=tools/utilities \
-  --exclude=opae-libs/include/opae/cxx/.clang-format \
-  --exclude=opae-libs/libopaecxx/.clang-format \
-  --exclude=opae-libs/pyopae/.clang-format \
-  --exclude=opae-libs/.clang-format \
-  --exclude=opae-libs/.clang-format \
-  -z -c -f opae.tar.gz opae-sdk
-
-mv opae.tar.gz "${RPMBUILD_DIR}/SOURCES"
+rm -rf _build
+mkdir _build
+cd _build
+cmake .. -DOPAE_BUILD_LEGACY=ON  -DOPAE_BUILD_TESTS=ON
+
+# Create source tarball.
+cd "${SCRIPT_DIR}"
+
+./create-tarball.sh opae-${version}
+
+mv ${SDK_DIR}/../opae-${version}.tar.gz "${RPMBUILD_DIR}/SOURCES"
+
+cp "${SDK_DIR}/opae.spec" ${SCRIPT_DIR}
 
 cd "${SCRIPT_DIR}"
 cp ./opae.spec "${RPMBUILD_DIR}/SPECS"
 
 ${SDK_DIR}/packaging/changelog.py rpm \
     --git-dir="${SDK_DIR}/.git" \
     --project="opae" \
-    --project-version="${PROJECT_VERSION}" \
+    --project-version="${version}" \
     >> "${RPMBUILD_DIR}/SPECS/opae.spec"
```

Comment:
This appends the latest changelog commits to the the end of the opae.spec copy which violates the requirement that the changelog be in reverse descending order.
We either need to insert the changelog commits in opae.spec or use the output of the this script to insert it in the appropriate location. I would recommend we do the former (committing it in opae.spec) instead of calling it in create. Doing it this way also allows for the creation of the RPM(s) depend only on the tarball and the .spec file itself.
It would looks something like the following (note, I had to manually set the release in this text as this changelog script only uses the version):

```
* Mon Dec 14 2020 The OPAE Dev Team <opae@lists.01.org> - 2.0.0-2
- fix code review comments
- fix  codeing style
- add create tarball script
- update fedora spec file and tarball generation script
```

Reply:
The idea was that there should be no changelog stored in opae.spec (the last thing in the file would be the %changelog line).  This way, we don't need to maintain super long changelogs, and we're not shipping changelogs for releases where they don't apply.

Reply:
fixed

Reply:
it's not fixed because this script is still appending changelog to the opae.spec copy in ${RPMBUILD_DIR}/SPECS.

Reply:
${RPMBUILD_DIR}/SPECS   add change log specific to PR

%changelog-> Release info, highlights of this release

Reply:
a change log is required in the fedora project so when this spec file is moved there at the least the release builds will need to be tracked

Reply:
fixed
Remove  ${RPMBUILD_DIR}/SPECS  from  RPM create file.
change log hard coded in opae.spec file